### PR TITLE
Include .analysis_options file into the list of analyzable files.

### PR DESCRIPTION
**Pull request by scheglov**
@pq implemented errors reporting for these files in DAS.
We need to show them in DPI.

### CODE REVIEW

**Comment by pq** [comment]
So awesome!  Needless to say, LGTM.

Note that in the future we can simplify this code a bit once server can tell clients what it cares about (https://github.com/dart-lang/sdk/issues/24746).

**Comment by alexander-doroshko** [comment]
I wanted to add a test to DartServerHighlightingTest but whatever I write DAS returns no errors (I used today's nightly SDK). Any idea?
Also I got:
```
Dart analysis server, SDK version 1.13.0-edge.50aa43569d892b3e395a812cdb2a42e66069c1be, server version 1.12.0, error: Error processing .analysis_options
OptionsFormatException: Expected ':'., <_FileSpan: from <FileLocation: 25 unknown source:3:10> to <FileLocation: 25 unknown source:3:10> "">
#0      AnalysisOptionsProvider.getOptionsFromString (package:analyzer/source/analysis_options_provider.dart:42)
#1      AnalysisOptionsProvider.getOptions (package:analyzer/source/analysis_options_provider.dart:20)
#2      ContextManagerImpl.processOptionsForContext (package:analysis_server/src/context_manager.dart:472)
#3      ContextManagerImpl._checkForAnalysisOptionsUpdate (package:analysis_server/src/context_manager.dart:736)
#4      ContextManagerImpl._handleWatchEvent (package:analysis_server/src/context_manager.dart:1169)
```

**Comment by scheglov** [comment]
For the following `.analysis_options` file content:
```
analyzer:
  errors:
    invalid-option: invalid-value
```
  I get two warnings:
image
image

**Comment by scheglov** [comment]
I've opened a new issue for the exception.
https://github.com/dart-lang/sdk/issues/24911

**Comment by alexander-doroshko** [comment]
@scheglov @pq DAS always returns no errors for me for the .analysis_options file with the same content for me. May be OS specific issue, I'm on Windows.
image

File: Dart/resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -159,6 +159,7 @@
     <annotator language="Dart" implementationClass="com.jetbrains.lang.dart.ide.annotator.DartColorAnnotator"/>
     <externalAnnotator language="Dart" implementationClass="com.jetbrains.lang.dart.analyzer.DartServerErrorsAnnotator"/>
     <externalAnnotator language="HTML" implementationClass="com.jetbrains.lang.dart.analyzer.DartServerErrorsAnnotator"/>
+    <externalAnnotator language="yaml" implementationClass="com.jetbrains.lang.dart.analyzer.DartServerErrorsAnnotator"/>
```
Comment (alexander-doroshko) [inline Dart/resources/META-INF/plugin.xml]: I moved this to dart-yaml.xml. YAML plugin might be not installed or or disabled, so it's better to keep yaml-dependent extensions in a corresponding xml that is parsed only if YAML plugin in enabled.

**Comment by pq** [comment]
> May be OS specific issue, I'm on Windows.

Ah.  This could be.  Possibly relevant: https://github.com/dart-lang/sdk/issues/24796.  I'll take a look when I'm back in the office later this week.

**Comment by alexander-doroshko** [comment]
I've filed https://github.com/dart-lang/sdk/issues/25114 to make sure it is not forgotten.

**Pull request merged**